src/wasm/wasm-binary.cpp

```diff
@@ -1211,7 +1212,31 @@ void WasmBinaryWriter::initializeDebugInfo() {
 }
 
 void WasmBinaryWriter::writeSourceMapProlog() {
-  *sourceMap << "{\"version\":3,\"sources\":[";
+  *sourceMap << "{\"version\":3,";
+
+  for (const auto& section : wasm->customSections) {
+    if (section.name == BinaryConsts::CustomSections::BuildId) {
+      U32LEB ret;
+      size_t pos = 0;
+      ret.read([&]() { return section.data[pos++]; });
+
+      if (section.data.size() != pos + ret.value) {
+        std::cerr
+          << "warning: build id section with an incorrect size detected!\n";
+        break;
+      }
+
+      *sourceMap << "\"debugId\":\"";
+      for (size_t i = pos; i < section.data.size(); i++) {
+        *sourceMap << std::setfill('0') << std::setw(2) << std::hex
+                   << +section.data[i];
```

Comment:
What is the `+` achieving here?

Reply:
that's just a conversion to integer (otherwise it'll print an ASCII representation of this character). However, I've updated it to use an explicit cast to not confuse anybody.